python/tvm/relax/frontend/onnx/onnx_frontend.py

```diff
@@ -1302,8 +1303,10 @@ def _impl_v18(cls, bb, inputs, attr, params):
         if scales is not None:
             assert isinstance(scales, relax.Constant), "Only constant scales currently supported."
             scales = scales.data.numpy()
-            sizes_shape = [dim.value for dim in x.struct_info.shape]
-            sizes = (sizes_shape * scales)[2:].astype("int64").tolist()
+            sizes = []
+            for i, dim in enumerate(x.struct_info.shape):
+                sizes.append(T.cast(scales[i] * dim, "int64"))
```

Comment:
It would be great to use `tir.cast` instead of `T.cast` as the later one is designed for TVMScript for most cases

Reply:
Change from T.cast to directly use cast which is imported from tvm.tir.generic